### PR TITLE
Changes to make ExperimentsRetentionMiddleware work with Django 1.10

### PR DESCRIPTION
**Pull request by ivanistheone**
This fixes the error reported in #154 but I'm not sure if it handles all edge cases, so consider this a first draft.

See https://docs.djangoproject.com/en/1.10/topics/http/middleware/#upgrading-pre-django-1-10-style-middleware for info.

This PR was used as inspiration: https://github.com/zestedesavoir/django-cors-middleware/compare/master...edmorley:support-django-110

### CODE REVIEW

**Comment by matclayton** [comment]
Closing this as we should support 1.10 and 1.11 now, we've run in production against both with the latest release.

**Comment by matclayton** [comment]
Re-opening it, as I'm an idiot, we don't support the 1.10 Middlewares yet, as we currently still run the old style.

File: experiments/middleware.py [hunk 1]
```diff
@@ -1,13 +1,33 @@
...
 
-class ExperimentsRetentionMiddleware(object):
     def process_response(self, request, response):
         # Don't track, failed pages, ajax requests, logged out users or widget impressions.
         # We detect widgets by relying on the fact that they are flagged as being embedable
         if response.status_code != 200 or request.is_ajax() or getattr(response, 'xframe_options_exempt', False):
             return response
 
+        # create session if not exists     see http://stackoverflow.com/a/5131421/127114
+        engine = import_module(settings.SESSION_ENGINE)
```
Comment (matclayton) [inline experiments/middleware.py]: I'm not sure this make a lot of sense, we've historically seen GoogleBot etc, create a lot of sessions and churn through the session store, so wouldn't want to create a session in such cases.

**Comment by matclayton** [comment]
I'm going to close this for now, and aim to merge the 1.10 middleware as part of #162 . If you can give me some context around the session changes I'm happy to consider them and figure out a way to solve whatever the issue is its causing.

File: experiments/middleware.py (hunk 1 above)
Comment (tysonclugg) [inline experiments/middleware.py]: You shouldn't have to worry about Googlebot creating lots of sessions - the optimisation belongs at a different layer, not here.

Considering that cookies are required to store the SESSION_ID, and that clearing the browser cache will clear said SESSION_ID, using [cookie-based sessions](https://docs.djangoproject.com/en/1.11/topics/http/sessions/#using-cookie-based-sessions) isn't such a bad idea.  You don't have to worry about storage, you reduce latency (not having a round trip to the DB/Cache to retrieve session data), and the "stale" sessions clear themselves.

Comment (matclayton) [inline experiments/middleware.py]: It makes a lot of sense, we however run db based sessions as there is a lot of data in them we use to aid on boarding and stuffing it in a cookie would require a lot of data transfer with the user :(

**Pull request closed without merging**